org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/utils/StringUtils.java

```diff
@@ -189,4 +189,31 @@ public static String urlDecode(String toDecode) {
   public static String xmlDecode(String toDecode) {
     return toDecode.replace("&quot;", "\"").replace("&apos;", "\'").replace("&lt;", "<").replace("&gt;", ">").replace("&amp;", "&");
   }
+
+  /**
+   *  When having a (possible multi-line) string and a (possible multi-line) substring, and the starting line of the
+   *  substring is required, this can be used to calculate it.
+   *
+   *  @param str that can contain subString
+   *  @param subString that is searched for
+   *  @return the line of the subString inside the main string
+   */
+  public static int getLineOfSubstring(final String str, final String subString) {
+    var subStringIndex = str.indexOf(subString);
+    if (subStringIndex == -1) {
+      return subStringIndex;
+    }
+
+    // In order to not have an ArithmeticException down below if there is no new line in the whole text!
+    if (!str.contains(lineSeperator)) {
+      return 0;
+    }
+
+    var preSubString = str.substring(0, subStringIndex);
+    return getNumberOfLines(preSubString);
+  }
+
+  public static int getNumberOfLines(final String str) {
+    return (str.length() - str.replace(lineSeperator, "").length()) / lineSeperator.length();
```

Comment:
Possible, easier-to-understand alternative

```
str.split(lineSeparator).length()
```
would it miss some cases?

Reply:
I wasn't even thinking about something as simple as this :face_in_clouds: It also shouldn't be missing anything as in case of no line separator in the string, there will be just one element in the list so length will be 1.

I also just noticed the other way around that when I refactored the upper method and moved this to a separate one I forgot to also move the null check in order to not divide by zero.
Once your review is done, I will take this into account and change it :smile: